src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultHostWorkspaceServicesProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class DefaultHostWorkspaceServicesProvider : HostWorkspaceServicesProvider
```

Comment:
In VSCode scenarios this will be used because VS will not have replaced this service to hoist its own workspace services. This means in VSCode scenarios we will still use MEF to resolve services for the AdhocWorkspace